core/src/toga/documents.py

```diff
@@ -16,7 +16,7 @@ def __init__(
         self,
         path: str | Path,
         document_type: str,
-        app: App = None,
+        app: App | None = None,
```

Comment:
Can `app` really be `None` here?

The implementation is missing several checks for `None` if that's the case.